src/PostgREST/QueryBuilder.hs

```diff
@@ -61,9 +61,22 @@ import           Data.Scientific         ( FPFormat (..)
 import           Prelude hiding          (unwords)
 import           PostgREST.ApiRequest    (PreferRepresentation (..))
 
+{-| The generic query result format used by API responses. The location header
+    is represented as a list of strings containing variable bindings like
+    @"k1=eq.42"@. If unused, it's null/Nothing rather than the empty list
+    because 'PostgreSQL.Binary.Decoder.arrayDimension' cannot decode an empty
+    array!
```

Comment:
Hi @nikita-volkov, we were doing a `SELECT array_agg(...)` and decoding it into `[ByteString]`. But sometimes the query will produce zero rows, so the aggregator function should return an empty array `{}::text[]`. I expected that to decode to the empty list, but instead it produces `failure "A missing dimension length"` in `PostgreSQL.Binary.Decoder.arrayDimension`.

In the commit shown here, we just worked around it by decoding to `Maybe [ByteString]` instead, and that works fine.

We're using `Hasql.Decoders` (`HD`) which in turn calls `PostgreSQL.Binary.Decoder`. Something like this:

```haskell
-- works, producing Maybe [ByteString]:
HD.nullableValue $ HD.array $ HD.arrayDimension replicateM $ HD.arrayValue HD.bytea
-- fails with "A missing dimension length" when aggregated result set is empty:
HD.value $ HD.array $ HD.arrayDimension replicateM $ HD.arrayValue HD.bytea
```